### DISCUSSION TITLE
Mistral training?

### DISCUSSION THREAD

**Discussion by caterpillarpants** [oobabooga/text-generation-webui#4322] [Ideas]
Can we train Mistral models? I'm not having success. Is this just a wait for transformers to get updated, or does real work need to be done on text-generation-webui?

I'm soooo pumped to train Mistral models and it's getting late, so I hope I'm not annoying.

`Training_PRO/script.py", line 708, in do_train
    target_modules=model_to_lora_modules[model_id],
KeyError: 'mistral'
`

**Comment by krslim** [comment]
I've had good results training ehartford/dolphin-2.1-mistral-7b
The latest version of Training_PRO from this repo includes mistral support. https://github.com/FartyPants/Training_PRO

**Reply by caterpillarpants** [reply]
awesome! I can't wait to try!

**Reply by FartyPants** [reply]
Really I didn't add mistral support, just fixed a bug that prevented mistral from training. The mistral support is in transformers so if it doesn't, update transformers.

**Reply by epolewski** [reply]
I'll echo that I was able to train it in Transformers.

**Comment by caterpillarpants** [comment]
I was able to get the training to work with a fresh install. I must have broken something trying to get deepspeed to work.

**Comment by Victorivus** [comment]
I am getting:

```
text-generation-webui/modules/training.py", line 345, in list_target_modules
    return model_to_lora_modules[model_id]
           ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
KeyError: 'mistral'
```

did you reinstall everything and just worked ? I haven't modified anything from the source code neither tried to get deepspeed working

**Reply by krslim** [reply]
I was getting this error with Training_PRO before FP fixed mistral support.

The extension was updated here, so with a fresh textgen install it should just work. I'm still able to train mistral with the transformers loader.

**Reply by Victorivus** [reply]
Yes, with the training PRO extension it works indeed, thanks. With the "basic" training tab it doesn't
